pkg/storegateway/bucket.go

```diff
@@ -2506,11 +2513,27 @@ func (r *bucketChunkReader) Close() error {
 	r.block.pendingReaders.Done()
 
 	for _, b := range r.chunkBytes {
-		r.block.chunkPool.Put(b)
+		r.block.chunkBytesPool.Put(b)
 	}
+	r.asyncPutChunks()
 	return nil
 }
 
+func (r *bucketChunkReader) asyncPutChunks() {
+	if len(r.results) == 0 {
+		return
+	}
+	go func() {
+		for _, res := range r.results {
+			for _, ent := range res {
+				for _, chk := range ent.chks {
+					r.block.chunksPool.put(chk.Raw)
+				}
+			}
+		}
+	}()
```

Comment:
Why is this asynchronous?

Reply:
removed @ 1bb10051a64c3bbded585b46d906394b9e58987c